lib/apiservers/engine/backends/volume.go

```diff
@@ -17,30 +17,70 @@ package vicbackends
 import (
 	"fmt"
 	"net/http"
+	"os"
 
 	derr "github.com/docker/docker/errors"
 
 	"github.com/docker/engine-api/types"
 	"github.com/vmware/vic/lib/apiservers/portlayer/client/storage"
+	"github.com/vmware/vic/lib/apiservers/portlayer/models"
+	"github.com/vmware/vic/lib/guest"
 	"github.com/vmware/vic/pkg/trace"
 )
 
+//Volume : struct which defines the docker personalities view of a Volume
 type Volume struct {
 	ProductName string
 }
 
+//Volumes : docker personality implementation for VIC
 func (v *Volume) Volumes(filter string) ([]*types.Volume, []string, error) {
 	return nil, make([]string, 0), fmt.Errorf("%s does not implement volume.Volumes", v.ProductName)
 }
 
+//VolumeInspect : docker personality implementation for VIC
 func (v *Volume) VolumeInspect(name string) (*types.Volume, error) {
 	return nil, fmt.Errorf("%s does not implement volume.VolumeInspect", v.ProductName)
 }
 
+//VolumeCreate : docker personality implementation for VIC
 func (v *Volume) VolumeCreate(name, driverName string, opts, labels map[string]string) (*types.Volume, error) {
-	return nil, fmt.Errorf("%s does not implement volume.VolumeCreate", v.ProductName)
+	defer trace.End(trace.Begin("Volume.VolumeCreate"))
+	result := &types.Volume{}
+
+	host, err := guest.UUID()
+	if host == "" {
+		host, err = os.Hostname()
+	}
+	if err != nil {
+		return result,
+			derr.NewErrorWithStatusCode(fmt.Errorf("Volume.CreateVolume got unexpected error getting hostname"),
```

Comment:
Yeah I was also going to advocate for that 👍 . The general errors we have from some of the other commands now really make it hard to know exactly what happened. I think we should look at expanding the responses of our swagger calls since our personality servers are really in charge of interpretting portlayer error codes to the user. So naturally I think we need more to know what is happening. LMKWYT